core/metacling/src/TCling.cxx

```diff
@@ -1263,6 +1263,11 @@ TCling::TCling(const char *name, const char *title)
       // Check that the gROOT macro was exported by any core module.
       assert(fInterpreter->getMacro("gROOT") && "Couldn't load gROOT macro?");
 
+      // When R binding is enabled and R related headers are loaded, these header
+      // files include Constants.h, which defines PI macro. It screw up every user's code
+      // containing PI.
+      fInterpreter->declare("#ifdef PI\n #undef PI\n #endif\n");
```

Comment:
I'm not sure if that's possible because this is not a name conflict caused by #include, but is caused by PI identifier became global by pre-loading R modules.